Improve mutations docs (#1206)
strawberry-graphql/strawberry: docs/general/mutations.md

# Mutations

As opposed to queries, mutations in GraphQL represent operations that modify
server-side data and/or cause side effects on the server. For example, you can have a
mutation that creates a user or a mutation that sends an email. Like in queries, they

> accept parameters, can return data or include nested fields as well.

Reviewer: ```
or include nested fields as well.
```
what do you mean by this? 😊

Author: That like queries you can write 
```
  {
    data {
       product {
          title
          reviews {
          ...
          }
      }
   }
```
Maybe is something obvious... 
is something the [GraphQL docs was mention 
](https://graphql.org/learn/queries/#mutations) so I thought to put something similar 
> Just like in queries, if the mutation field returns an object type, you can ask for nested fields.

# Mutations

As opposed to queries, mutations in GraphQL represent operations that modify server-side
data and/or cause side effects on the server. For example, you can have a mutation that

> adds a book instance or a mutation that sends an email. Like in queries, they accept

Reviewer 2: ```suggestion
creates a new instance in your application or a mutation that sends an email. Like in queries, they accept
```

> parameters, can return an object type and can include nested fields as well. This can be

Reviewer 2: ```suggestion
parameters and can return anything a regular field can, including new types and existing object types. This can be
```

@strawberry.type
class Mutation:
    @strawberry.mutation
    def add_book(self, title: str, author: str) -> Book:
        print(f'Adding {title} by {author}')

        return Book(title=title, author=author)

schema = strawberry.Schema(query=Query, mutation=Mutation)
```

Like queries, mutations are defined in a class that is then passed to the Schema

> function. Here we create a `addBook` mutation that accept a title and an author and

Reviewer 2: ```suggestion
function. Here we create an `addBook` mutation that accepts a title and an author and
```

returns a `Book` type.

We would send the following GraphQL document to our server to execute the mutation:

```graphql

> addBook(title: "The Little Prince", author: "Antoine de Saint-Exupéry")

Reviewer 2: ```suggestion
mutation {
  addBook(title: "The Little Prince", author: "Antoine de Saint-Exupéry") {
    title
  }
}
```

```

> The `addBook` mutation is a simplified example. In a real-world application we usually
> want to return more information if an error occurs.

Reviewer 2: ```suggestion
The `addBook` mutation is a simplified example. In a real-world application mutations will often need to handle errors and communicate those errors back to the client. For example we might want to return an error if the book already exists.
```

You can checkout our documentation on

> [dealing with errors](/docs/guides/errors#expected-errors) to learn how to return a
> union of types in mutation.

Reviewer 2: ```suggestion
[dealing with errors](/docs/guides/errors#expected-errors) to learn how to return a
union of types from a mutation.
```

sendEmail(email: "patrick@example.org")
```

> The `sendEmail` mutation is a simplified example. However, in a real-world
> application, we usually want to return more information if an error occurs.
> Refer to [Dealing with errors](/docs/guides/errors#expected-errors) documentation to
> learn how to return a union of types in mutation.

Reviewer: ```suggestion
The `sendEmail` mutation is a simplified example. In a real-world
application we usually want to return more information if an error occurs.

You can checkout our documentation on [dealing with errors](/docs/guides/errors#expected-errors) to
learn how to return a union of types in mutation.
```
